File: Toy_Language_Interpreter/Model/Statements/AssignStatement.java
package Model.Statements;

import ADTs.IDictionary;
import Exceptions.*;
import Heap.IHeap;
import Model.Expressions.IExpression;
import Model.ProgramState;

public class AssignStatement implements IStatement {
    private String variable;
    private IExpression exp;

    public AssignStatement(String var, IExpression ex) {
        this.variable = var;
        this.exp = ex;
    }

    public String toString() {
        return this.variable + " = " + this.exp.toString();
    }

    public ProgramState execute(ProgramState state) throws UndefinedVariableException, DivisionByZeroException, UndefinedOperationException, HeapReadingException {
        IDictionary<String, Integer> symTable = state.getSymTable();
        IHeap heap = state.getHeap();
        int result = this.exp.eval(symTable, heap);
        symTable.put(variable, result);
        return state;
    }
}

File: Toy_Language_Interpreter_C/Statements/AssignStatement.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace a7
{
    class AssignStatement : IStatement
    {
        private String variable;
        private IExpression expression;

        public AssignStatement(String var, IExpression exp)
        {
            this.variable = var;
            this.expression = exp;
        }

        public ProgramState Execute(ProgramState state)
        {
            try
            {
                int result = this.expression.Eval(state.SymTable);
                state.SymTable.Put(variable, result);
                return state;
            }
            catch (DivideByZeroException)
            {
                throw new DivideByZeroException("DIVISION BY ZERO. Cannot divide by 0.");
            }
            catch(UndefinedVariableException)
            {
                throw new UndefinedVariableException(this.variable);
            }
        }

        public override string ToString()
        {
            return this.variable + " = " + this.expression.ToString();
        }
    }
}
